bot/exts/utils/extensions.py

```diff
@@ -0,0 +1,196 @@
+        await ctx.send_help(ctx.command)
+
+    @extension_group.command(name="load", aliases=["l"])
+    async def loading(self, ctx: commands.Context, *extensions: Extension) -> None:
+        """Loads the given extensions.
+        If `*` is provided as the extension then loads every single extension.
+        To reload the extension of an entire folder, `**` must be before the folder name.
+        """
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+                        
+        if '*' in extensions:
+            extensions = e.EXTENSIONS
+
+        messages = self.manage("load", extensions)
+        await ctx.channel.send(messages)
+
+    @extension_group.command(name="reload", aliases=["rl"])
+    async def reloading(self, ctx: commands.Cog, *extensions: Extension) -> None:
+        """
+        Reloads the given extensions.
+        If `*` is provided as the extension then reloads all the extensions.
+        To reload the extension of an entire folder, `**` must be before the folder name.
+        """
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+        
+        if '*' in extensions:
+            extensions = e.EXTENSIONS
+
+        messages = self.manage("reload", extensions)
+        await ctx.channel.send(messages)
+
+    @extension_group.command(name="unload", aliases=["ul"])
+    async def unloading(self, ctx: commands.Cog, *extensions: Extension) -> None:
+        """
+        Unloads the given extensions.
+        If `*` is provided as the extension then unloads every all the extensions.
+        To reload the extension of an entire folder, `**` must be before the folder name.
+        """
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+        
+        if '*' in extensions:
+            extensions = [  ext for ext in e.EXTENSIONS 
+                            if ext not in UNLOAD_BLACKLIST
+                        ]
+
+        blacklists = [
+            blacklisted_extesnion
+            for blacklisted_extesnion in extensions
+            if blacklisted_extesnion in UNLOAD_BLACKLIST
+        ]
+        
+        if blacklists:
+            await ctx.channel.send(
+                f":x: `{' ,'.join(blacklists)}` extensions may not be unloaded"
+            )
+            logger.info(
+                f"{str(ctx.author)} Tried to unload blacklisted extension"
+                f"{', '.join(blacklists)}"
+            )
+            return
+
+        messages = self.manage("unload", extensions)
+        await ctx.channel.send(messages)
+    
+
+    def manage(
+        self,
+        action: str,
+        extensions: t.Tuple[str],
+    ) -> str:
+        """Apply an action to the provided extension/extensions."""
+        if not extensions:
+            raise ValueError("Extension not provided.")
+
+        if all(actions != action for actions in ["load", "unload", "reload"]):
```

Comment:
I'm not sure what you are trying to do here.
Wouldn't
```suggestion
        if  action not in ["load", "unload", "reload"]:
```
suffice?

Reply:
Ah right.